### PR TITLE
Implement different success processing for pre-auth and non-pre-auth orders

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
There is a redundant order/save call from the non pre-auth flow still present when pre-auth is turned on. It throws and error on HUF. Removing it fixes the issue.

Removing the call via configuration option, Pre-Auth order creation.

Fixes:
https://app.asana.com/0/941920570700290/1133661540833258/f

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] I have created or modified unit tests to sufficiently cover my changes.

### CODE REVIEW

File: Helper/Config.php [hunk 1]
```diff
@@ -838,6 +843,22 @@ private function getAdditionalConfigProperty($name, $storeId = null)
...
     }
 
+    /**
+     * Get Is Pre-Auth flag from config
+     *
+     * @param int|string|Store $store
+     *
+     * @return  boolean
+     */
+    public function getIsPreAUth($store = null)
```
Comment (matt-thomason) [inline Helper/Config.php]: Weird capitalization here `AUth`

File: view/adminhtml/templates/boltpay/js/replacejs.phtml [hunk 2]
```diff
@@ -376,21 +376,41 @@ require([
         },
 
         success: function (transaction, callback) {
+
+            /**
+             * Success transaction handler.
+             * Sets the success url for the non-preauth flow.
```
Comment (matt-thomason) [inline view/adminhtml/templates/boltpay/js/replacejs.phtml]: Isn't this being called if `is_pre_auth` is true?

File: view/frontend/templates/js/replacejs.phtml [hunk 3]
```diff
@@ -553,6 +553,34 @@ ob_start();
             },
 
             success: function (transaction, callback) {
+                /**
+                 * Success transaction handler.
+                 * Sets the success url for the non-preauth flow.
```
Comment (matt-thomason) [inline view/frontend/templates/js/replacejs.phtml]: Same here as well.

File: etc/config.xml [hunk 4]
```diff
@@ -56,6 +56,7 @@ tr.shipping.totals td.amount span.price,
                 <bolt_order_caching>1</bolt_order_caching>
                 <api_emulate_session>1</api_emulate_session>
                 <should_minify_javascript>1</should_minify_javascript>
+                <is_pre_auth>0</is_pre_auth>
```
Comment (daisy1754) [inline etc/config.xml]: can we make it true by default?

File: view/adminhtml/templates/boltpay/js/replacejs.phtml (hunk 2 above)
Comment (danac-gs) [reply view/adminhtml/templates/boltpay/js/replacejs.phtml]: @matt-thomason 

If pre-auth is true then the redirect_url, which is created and returned in result of create order call, is implicitly opened from the checkout, when passed callback() is executed. And it redirects to the success page if passes validation.

In no pre-auth we need to call order/save, which returns the success url that we save as a property of the callbacks object. on the success order/save call, as the last step, after storing the success_url, the callback() is called. One of the things it does is closing the checkout modal. In the close event handler we check for the success_url and open the success page.

To sumarize:
pre-auth - checkout implicitly does the success page redirect
no pre-auth - the redirect is called explicitly in the plugin code

The later was already done in pre-auth and it was redundant.

File: etc/config.xml (hunk 4 above)
Comment (danac-gs) [reply etc/config.xml]: @daisy1754

It is deliberately set as false. Two reasons:

1. When set to 0 It actually works both ways, pre-auth and no pre-auth. At least in most of the cases. If not for HUF we wouldn't notice it yet.

2. If the merchant updates the plugin on their own (as Capillus did today) without turning on the pre-auth flag and setting the create order endpoints we loose backward compatibility if the default is true.

Comment (daisy1754) [reply etc/config.xml]: okay fair

File: view/frontend/templates/js/replacejs.phtml (hunk 3 above)
Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml]: Same as above.

File: etc/config.xml (hunk 4 above)
Comment (daisy1754) [reply etc/config.xml]: @matt-thomason let's make sure for new merchants we instruct to set this to true

File: Helper/Config.php (hunk 1 above)
Comment (danac-gs) [reply Helper/Config.php]: fixed

**Pull request merged**